recipe/meta.yaml

```diff
@@ -9,33 +9,69 @@ source:
   sha256: 7756cf4b4a427a856137c40c3fdf27422fcf546249b70f9ba4c213ef1fbee051
 
 build:
-  number: 1
+  number: 2
   skip: True  # [win and vc<14]
 
 requirements:
   build:
-    - cmake
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
-  host:
-    - eigen
-    - nlopt
-    - boost-cpp
-    - tbb
-    - tbb-devel
-    - ipopt               # [unix]
-  run:
-    - eigen
-    - nlopt
-    - boost-cpp
-    - tbb
-    - tbb-devel
-    - ipopt               # [unix]
-
-test:
-  commands:
-    - test -f $PREFIX/include/pagmo/pagmo.hpp               # [unix]
-    - if not exist %LIBRARY_INC%\\pagmo\\pagmo.hpp exit 1   # [win]
+    - cmake
+
+outputs:
+  - name: pagmo
+    requirements:
+      host:
+        - eigen
+        - nlopt
+        - boost-cpp
+        - tbb-devel
+        - ipopt  # [unix]
+      run:
+        # NOTE: nlopt and tbb
+        # use run_exports().
+        - {{ pin_compatible('boost-cpp', max_pin='x.x') }}
+        - ipopt  # [unix]
+    files:
+      - lib/libpagmo*{{ SHLIB_EXT }}   # [not win]
+      - lib/libpagmo.so.*              # [linux]
+      - lib/libpagmo.*.dylib           # [osx]
+      - Library/bin/pagmo.dll          # [win]
+    test:
+      commands:
+        - test -f $PREFIX/lib/libpagmo.so                                     # [linux]
+        - test -f $PREFIX/lib/libpagmo.dylib                                  # [osx]
+        - if not exist %PREFIX%\\Library\\bin\\pagmo.dll exit 1               # [win]
+
+  - name: pagmo-devel
+    build:
+      run_exports:
+        # API/ABI change at minor rev bumps.
+        - {{ pin_subpackage('pagmo', max_pin='x.x') }}
```

Comment:
If pagmo headers use `tbb` headers, then `tbb` should be added as a `run_exports`.

Reply:
Thanks for the review!

pagmo does not have any ``tbb`` include in its public headers, but the API does make use of some headers from ``nlopt`` and ``ipopt``. Should I add these to ``run_exports``? Could you elaborate on what is the general guideline on what should be in the ``run_exports`` of a ``-devel`` package?

Reply:
Or, to put it in other terms, what would be the danger in not having ``tbb`` in ``run_exports`` if the pagmo public API did indeed use ``tbb`` headers?

Reply:
I was re-reading the conda docs, is this because of the non-transitive nature of ``run_exports``?

> run_exports only affects directly named dependencies. For example, if you have a metapackage that includes a compiler that lists run_exports, you also need to define run_exports in the metapackage so that it takes effect when people install your metapackage. This is important, because if run_exports affected transitive dependencies, you would see many added dependencies to shared libraries where they are not actually direct dependencies. For example, Python uses bzip2, which can use run_exports to make sure that people use a compatible build of bzip2. If people list python as a build time dependency, bzip2 should only be imposed for python itself, and should not be automatically imposed as a runtime dependency for the thing using python.

https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html

Reply:
Yes, it's because it is non-transitive